path_login.go

```diff
@@ -262,35 +319,92 @@ func (b *backend) getCallerIdentity(header http.Header, rawURL string) (*sts.Get
 
 	response, err := b.identityClient.Do(request)
 	if err != nil {
-		return nil, errwrap.Wrapf("error making request: {{err}}", err)
+		return nil, fmt.Errorf("error making request: %w", err)
 	}
 	defer response.Body.Close()
 
 	if response.StatusCode != 200 {
 		b, err := ioutil.ReadAll(response.Body)
 		if err != nil {
-			return nil, errwrap.Wrapf("error reading response body: {{err}}", err)
+			return nil, fmt.Errorf("error reading response body: %w", err)
 		}
 		return nil, fmt.Errorf("received %d checking caller identity: %s", response.StatusCode, b)
 	}
 
 	result := &sts.GetCallerIdentityResponse{}
 	if err := json.NewDecoder(response.Body).Decode(result); err != nil {
-		return nil, errwrap.Wrapf("error decoding response: {{err}}", err)
+		return nil, fmt.Errorf("error decoding response: %w", err)
 	}
 	return result, nil
 }
 
-func getSTSEndpoint() (string, error) {
-	r := &endpoints.LocalGlobalResolver{}
-	endpoint, supported, err := r.TryResolve(&endpoints.ResolveParam{
-		Product: "sts",
-	})
+// generateLoginData is a helper that returns a new signed GetCallerIdentity request.
+func generateLoginData(req *logical.Request, data *framework.FieldData) (*framework.FieldData, error) {
+	accessKey := data.Get("access_key").(string)
```

Comment:
We could use `data.GetOk` instead as a safer access approach, but this is definitely not required, since consistency with existing code is also valuable, and it's not a reasonable ask to do this to the whole project. Still, thought it was worth noting for consideration.

Reply:
I think it makes sense to use GetOk instead. Thanks!

Reply:
Actually, I think I will stick with Get. GetOk will make it slightly more verbose. But more importantly GetOk will change the behavior (no longer panic if the field is not in FieldData). And I don't think it is worth changing all Get calls in the plugin.

Reply:
Yeah, it's a lot of work to change all of the calls in the plugin. Sounds good!